consensus/ibft/consensus_backend.go

```diff
@@ -0,0 +1,319 @@
+package ibft
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/consensus"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+func (i *backendIBFT) BuildProposal(blockNumber uint64) []byte {
+	var (
+		latestHeader      = i.blockchain.Header()
+		latestBlockNumber = latestHeader.Number
+	)
+
+	if latestBlockNumber+1 != blockNumber {
+		return nil
+	}
+
+	snap := i.getSnapshot(latestBlockNumber)
+	if snap == nil {
+		i.logger.Error("cannot find snapshot", "num", latestBlockNumber)
+
+		return nil
+	}
+
+	block, err := i.buildBlock(snap, latestHeader)
+	if err != nil {
+		i.logger.Error("cannot build block", "num", blockNumber, "err", err)
+
+		return nil
+	}
+
+	return block.MarshalRLP()
+}
+
+func (i *backendIBFT) InsertBlock(proposal []byte, committedSeals [][]byte) {
```

Comment:
This function is also. We need to think about where we call log.

Reply:
Please see:

https://github.com/0xPolygon/polygon-edge/pull/650#discussion_r933090429